src/main/java/bio/terra/service/resourcemanagement/google/GoogleBillingService.java

```diff
@@ -73,38 +91,32 @@ private static Cloudbilling cloudbilling() {
     @Override
     public boolean canAccess(BillingProfile billingProfile) {
         String accountId = "billingAccounts/" + billingProfile.getBillingAccountId();
+        ResourceName resource = BillingAccountName.of(accountId);
         List<String> permissions = Collections.singletonList("billing.resourceAssociations.create");
-        TestIamPermissionsRequest permissionsRequest = new TestIamPermissionsRequest().setPermissions(permissions);
+        TestIamPermissionsRequest permissionsRequest = TestIamPermissionsRequest.newBuilder()
+            .setResource(resource.toString())
+            .addAllPermissions(permissions)
+            .build();
         try {
-            Cloudbilling.BillingAccounts.TestIamPermissions testIamPermissions =
-                cloudbilling().billingAccounts().testIamPermissions(accountId, permissionsRequest);
-            List<String> actualPermissions = testIamPermissions.execute().getPermissions();
-            return actualPermissions != null && actualPermissions.equals(permissions);
-        } catch (GoogleJsonResponseException e) {
-            int status = e.getStatusCode();
-            if (status == 400 || status == 404) {
-                // The billing account id is invalid or does not exist. This counts as inaccessible.
-                return false;
-            }
-            String message = String.format("%s Error, Could not check permissions on: %s", status, accountId);
-            throw new BillingServiceException(message, e);
-        } catch (IOException e) {
-            String message = String.format("Could not check permissions on: %s", accountId);
+            TestIamPermissionsResponse response = cloudbilling().testIamPermissions(permissionsRequest);
+            return response.getPermissionsList().containsAll(permissions);
```

Comment:
It seems like instead of returning a JSON response, these APIs now use GRPC instead, so I am not sure what the equivalent code might look like to identify status.

Reply:
I do not know. The expression you wrote makes sense to me.